src/config.h

```diff
@@ -293,7 +297,7 @@ void setproctitle(const char *fmt, ...);
 #include <kernel/OS.h>
 #define redis_set_thread_title(name) rename_thread(find_thread(0), name)
 #else
-#if (defined __APPLE__ && defined(MAC_OS_X_VERSION_10_7))
+#if (defined __APPLE__ && defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 1070)
```

Comment:
Could you please define `MAC_OS_10_7_DETECTED` at the start of this file, similar to `MAC_OS_10_6_DETECTED`.

Being consistent always helps, Thanks!

Please add a comment to, this will greatly help in future to tackle similar problems if observed.

Reply:
It's only used in one place, so it's not the same as MAC_OS_10_6_DETECTED which is used in more than three places.

Reply:
I'm okay, but remember it's not about one place or three places. It's about consistency and that always helps in the long run. Tomorrow the issue could arrive for another versions, and if we maintain all such things at a single place really helps. Ideally we should have a separate header file for maintaining such things. Thanks!

Reply:
@shahsb https://github.com/redis/redis/pull/12611#discussion_r1339849033
this is the reason why we don't put it in `fmacros.h`